pkg/cloud/openstack/machine/machineScript.go

```diff
@@ -81,7 +81,7 @@ func nodeStartupScript(cluster *clusterv1.Cluster, machine *clusterv1.Machine, t
 		MachineSpec:    machineSpec,
 		PodCIDR:        getSubnet(cluster.Spec.ClusterNetwork.Pods),
 		ServiceCIDR:    getSubnet(cluster.Spec.ClusterNetwork.Services),
-		MasterEndpoint: getEndpoint(cluster.Status.APIEndpoints[0]),
+		MasterEndpoint: masterEndpoint,
```

Comment:
We can create a `GetMasterEndpoint` function, pass it to the template and then have the templates under [`examples/openstack`](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/tree/master/cmd/clusterctl/examples/openstack) access it with `{{ .GetMasterEndpoint }}`